lib/ukrand/chacha.c

```diff
@@ -137,7 +137,27 @@ void uk_swrand_init_r(struct uk_swrand *r, unsigned int seedc,
 	r->k = 16;
 }
 
-__u32 uk_swrand_randr_r(struct uk_swrand *r)
+static int uk_chacha_init(struct uk_init_ctx *ictx __unused)
+{
+	unsigned int seedc = 10;
+	__u32 seedv[10];
+	unsigned int i;
+	int ret;
+
+	uk_pr_info("Initialize random number generator...\n");
+
+	for (i = 0; i < seedc; i++) {
+		ret = ukarch_random_seed_u32(&seedv[i]);
+		if (unlikely(ret))
+			uk_pr_warn("Could not generate random seed\n");
```

Comment:
>  to also directly return the error code.

This implies both, yes :grin: 

> Also, I am not sure what happens if an init function fails.

This will be propagated to ukboot so Unikraft init will fail so all fine.